nni/retiarii/experiment/config/experiment_config.py

```diff
@@ -28,6 +28,7 @@ def _get_ee_config_class(engine_name):
 @dataclass(init=False)
 class RetiariiExeConfig(ExperimentConfig):
     # FIXME: refactor this class to inherit from a new common base class with HPO config
+    experiment_type: str = 'nas'
```

Comment:
I think this field should be created at dump. How do you make sure every subclass of ExperimentConfig had this field. Imagine, what if users accidentally change this field?

Reply:
Could we put `experiment_type` into `.experiment`, I think this is experiment `metadata`.

Reply:
> I think this field should be created at dump. How do you make sure every subclass of ExperimentConfig had this field. Imagine, what if users accidentally change this field?

Similar to `TrainingServiceConfig.platform`, the type should be `Literal['nas']` so `validate()` will check it.

Reply:
> Could we put `experiment_type` into `.experiment`, I think this is experiment `metadata`.

I prefer not. I think `.experiment` is mainly for NNI manager and NNI manager should not care about this.

Reply:
> > Could we put `experiment_type` into `.experiment`, I think this is experiment `metadata`.
> 
> I prefer not. I think `.experiment` is mainly for NNI manager and NNI manager should not care about this.

these metadata are written by python side, when we view or resume an experiment, we must read `.experiment`, so I think `.experiment` is somehow also for python side control the experiment.

Reply:
> > > Could we put `experiment_type` into `.experiment`, I think this is experiment `metadata`.
> > 
> > 
> > I prefer not. I think `.experiment` is mainly for NNI manager and NNI manager should not care about this.
> 
> these metadata are written by python side, when we view or resume an experiment, we must read `.experiment`, so I think `.experiment` is somehow also for python side control the experiment.

When `experiment_type` is a field of experiment config, NNI manager can also obtain the experiment type by parsing experiment config, right? not necessarily in `.experiment`. And, `experiment_type` is a property of an experiment, putting it in experiment config allows everyone (who has access to the experiment config) to easily get to know the experiment type.